pkg/kafka/event_stream_writer.go

```diff
@@ -29,11 +29,6 @@ type EventEnvelope struct {
 	Metadata interface{} `json:"metadata"`
 }
 
-type EventStreamWriter interface {
-	Write(ctx context.Context, eventName string, key []byte, payload interface{}) error
-	Close()
-}
-
 type KafkaConfig struct {
```

Comment:
I'd also consider renaming these so that we don't have `kafka.Kafka...` everywhere.